### DISCUSSION TITLE
can I upload an image from client to server?

### DISCUSSION THREAD

**Discussion by AndyJado** [leptos-rs/leptos#2215] [Q&A] [answered]
there are only buttons and text in examples

**Comment by gbj** [comment, marked as answer]
Yes.

The answer to the question "How?" depends a lot on how much you already know about the various Web APIs involved, how comfortable you are accessing web APIs from Rust using the `web-sys` crate, etc.

Here's an example another user created that I found by searching in our Discord -- I don't take any credit for it.
```rust
#[component]
fn UploadFileForm<F>(on_upload_finished: F) -> impl IntoView
where
    F: Fn(bool) + 'static + Clone,
{
    let error: RwSignal<Option<Box<dyn Error>>> = create_rw_signal(None);
    let form_ref: NodeRef<html::Form> = create_node_ref();
    let on_submit = move |ev: SubmitEvent| {
        ev.prevent_default();
        let form = form_ref.get().expect("noform");
        let action = form
            .get_attribute("action")
            .unwrap_or_default()
            .to_lowercase();
        let form_data = web_sys::FormData::new_with_form(&form).unwrap_throw();
        spawn_local(async move {
            let res = gloo_net::http::Request::post(&action)
                .header("Accept", "application/json")
                .body(form_data)
                .unwrap()
                .send()
                .await;
            if let Err(err) = res {
                log::error!("<Form/> error while POSTing: {err:#?}");
                error.try_set(Some(Box::new(err)));
            }
        });
        let has_error = error.with_untracked(|value| value.is_some());
        on_upload_finished(has_error);
    };
    view! {
        <Form error=error node_ref=form_ref enctype="multipart/form-data".to_string() action="/upload" on:submit=on_submit>
            <label>"Upload file" <input type="file" id="path" name="path"/></label>
            <input type="submit" value="Upload"/>
        </Form>
    }
}
```

Alternately, with the new server functions on the main branch we have examples of file upload using multipart form data

https://github.com/leptos-rs/leptos/blob/7d1ce45a579523e918de7f784e1452a8e1085f88/examples/server_fns_axum/src/app.rs#L321-L384

**Reply by AndyJado** [reply]
it seems I couldn't read the Request Body now: `RequestParts` just disappered.
```rust
    let req_parts = use_context::<leptos_axum::RequestParts>();
```

**Reply by gbj** [reply]
> it seems I couldn't read the Request Body now: `RequestParts` just disappered.

When you are looking for help it is really helpful to include more information than you have. For example, are you using 0.6/the main branch, or 0.5? What does "just disappeared" mean (is there a compile error?) etc.

It may be as simple as replacing this with [`Parts` ](https://docs.rs/http/latest/http/request/struct.Parts.html); we have removed `RequestParts` as no longer necessary because `http::Parts` is now `Clone`.
